docs/enhancements/clusterpool-inventory.md

```diff
@@ -0,0 +1,180 @@
+# Clusterpool for on-prem cloud providers
+
+[HIVE-1367](https://issues.redhat.com/browse/HIVE-1367)
+
+- [Clusterpool for on-prem cloud providers](#clusterpool-for-on-prem-cloud-providers)
+  - [Summary](#summary)
+  - [Problem Statement](#problem-statement)
+  - [Proposal](#proposal)
+    - [Summary](#summary-1)
+    - [`ClusterPool.Spec.Inventory`](#clusterpoolspecinventory)
+    - [How To Use](#how-to-use)
+    - [Validation](#validation)
+    - [`Size` and `MaxSize`](#size-and-maxsize)
+    - [Pool Version](#pool-version)
+    - [Handling Inventory Updates](#handling-inventory-updates)
+      - [Adding An Inventory](#adding-an-inventory)
+      - [Adding An Entry to the Inventory](#adding-an-entry-to-the-inventory)
+      - [Removing An Entry from the Inventory](#removing-an-entry-from-the-inventory)
+      - [Deleting The Inventory](#deleting-the-inventory)
+    - [Maintaining the lease of the InstallConfigPatch](#maintaining-the-lease-of-the-installconfigpatch)
+    - [Fairness](#fairness)
+        
+## Summary
+
+As a cluster administrator, I want to be able to use ClusterPools to provision clusters on on-prem clouds like vSphere, OpenStack and Baremetals.
+
+## Problem Statement
+Provisioning a cluster requires configuration unique to that cluster.
+For example, the cluster's name is used to build the hostnames for the API and console.
+_Someone_ has to handle the DNS that resolves those hostnames to the IPs of the services created by the installer.
+One option is to create your ClusterDeployment with `ManageDNS=True` and we'll create a DNSZone on the fly.
+
+But that doesn't work for some cloud providers such a vSphere, OpenStack and Baremetals.
+
+In such cases, one solution is to configure DNS manually and then create your ClusterDeployment with the right `Name` and `BaseDomain` so that the assembled hostnames match those entries. This is fine if you're creating ClusterDeployments by hand, but breaks down if ClusterDeployment names are being generated with random slugs, as is the case with ClusterPools.
+
+## Proposal
+### Summary
+Allow ClusterPool to accept an inventory of install config json patches to be used when generating ClusterDeployments for the pool.
+
+### `ClusterPool.Spec.Inventory`
+Add a field to `ClusterPool.Spec` called `Inventory`.
+
+It has an array of object references to InstallConfigPatch custom resource. This CR will have a base64 encoded json patch (RFC 6902) for default install config generated by clusterpool controller. The json patch will have cluster specific changes, for vSphere this corresponds to configured DNS hostnames
```

Comment:
Let's state that the patches will be applied in the order listed in the inventory.

Reply:
Added on line 92